### DISCUSSION TITLE
[Coil3] HTTP headers : Confused by the documentation

### DISCUSSION THREAD

**Discussion by RobbWatershed** [coil-kt/coil#2558] [Q&A] [answered]
I'd like to set cookies through HTTP headers for individual requests.

[That page](https://coil-kt.github.io/coil/recipes/#headers) explains how, but I realize Coil3 does not provide `ImageRequest.Builder.setHeader` anymore 😦 

I dug up a vague reference to the `NetworkFetcher` class in #2474 but my small brain can't bridge the gaps.

=> Could someone provide a sample code to illustrate that use case?

Thanks in advance~

**Comment by colinrtwhite** [comment, marked as answer]
Sorry, that recipes page needs to be updated. Here's an example of how to set headers for a request in Coil 3.0:

```kotlin
val headers = NetworkHeaders.Builder()
    .set(key, value)
    .build()
val request = ImageRequest.Builder(context)
    .data(url)
    .httpHeaders(headers)
    .build()

**Reply by RobbWatershed** [reply]
Perfect - just what I needed. Thanks man.
